src/templates/next/layouts/Collection/Collection.tsx

```diff
@@ -1,15 +1,97 @@
+import { useState } from "react"
 import type { CollectionPageSchema } from "~/engine"
 import { CollectionCard } from "../../components"
 import { Heading } from "../../typography/Heading"
 import { Paragraph } from "../../typography/Paragraph"
 import { Skeleton } from "../Skeleton"
+import type { Filter as FilterType, AppliedFilter } from "../../types/Filter"
+import { CollectionSearch, Filter, Pagination } from "../../components/shared"
+
+const extractCategories = (
+  items: CollectionPageSchema["page"]["items"],
+): Record<string, number> => {
+  const categories: Record<string, number> = {}
+  items.forEach((item) => {
+    if (item.category in categories) {
+      categories[item.category] += 1
+    } else {
+      categories[item.category] = 1
+    }
+  })
+  return categories
+}
+
+const getFilters = (categories: Record<string, number>): FilterType[] => {
+  return [
+    {
+      id: "category",
+      label: "Category",
+      items: Object.entries(categories).map(([label, count]) => ({
+        id: label.toLowerCase(),
+        label,
+        count,
+      })),
+    },
+  ]
+}
+
+const getFilteredResults = (
+  items: CollectionPageSchema["page"]["items"],
+  appliedFilters: AppliedFilter[],
+  search: string,
+  itemsPerPage: number,
+  currPage: number,
+): CollectionPageSchema["page"]["items"] => {
+  const filteredItems = items.filter((item) => {
+    if (
+      search !== "" &&
+      !item.title.toLowerCase().includes(search.toLowerCase()) &&
+      !item.description.toLowerCase().includes(search.toLowerCase())
+    ) {
+      return false
+    }
+
+    for (const filter of appliedFilters) {
+      if (filter.id === "category") {
```

Comment:
abit unclear why this magic string works - i know that this is via what we specify here but maybe we should make it a const in this file?